src/explore-education-statistics-common/src/components/form/rhf/__tests__/RHFForm.test.tsx

```diff
@@ -0,0 +1,491 @@
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
+    });
+  });
+
+  test('does not re-focus error summary when changing input', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: '',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm id="test-form" showSubmitError onSubmit={noop}>
+              <label htmlFor="firstName">First name</label>
+              <input
+                id="firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
+    });
+    const input = screen.getByLabelText('First name');
+
+    input.focus();
+
+    fireEvent.change(input, {
+      target: {
+        value: 'a first name',
+      },
+    });
+    // await userEvent.type(screen.getByLabelText('First name'), 'a first name');
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).not.toHaveFocus();
+    });
+  });
+
+  test('re-focuses error summary on re-submit', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: '',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm id="test-form" showSubmitError onSubmit={noop}>
+              <label htmlFor="firstName">First name</label>
+              <input
+                id="firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
+    });
+
+    screen.getByLabelText('First name').focus();
+    expect(screen.getByRole('alert')).not.toHaveFocus();
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
+    });
+  });
```

Comment:
As earlier, using `getByRole` in `waitFors`. We can optimise this with `getByText`